src/decoder/mod.rs

```diff
@@ -1107,6 +1395,41 @@ impl<R: Read + Seek> Decoder<R> {
         Ok(())
     }
 
+    fn init_tile_attributes(&mut self, buffer_byte_len: usize) -> TiffResult<()> {
+        if self.tile_attributes.is_none() {
+            let tile_width = usize::try_from(self.get_tag_u32(Tag::TileWidth)?)?;
+            let tile_length = usize::try_from(self.get_tag_u32(Tag::TileLength)?)?;
+
+            let tiles_across = (usize::try_from(self.width)? + tile_width - 1) / tile_width;
+            let tiles_down = (usize::try_from(self.height)? + tile_length - 1) / tile_length;
+
+            let samples_per_pixel = self.bits_per_sample.len();
+
+            let tile_sample_count = tile_length * tile_width * samples_per_pixel;
+            let padding_right = (tiles_across * tile_width) - usize::try_from(self.width)?;
+            let tile_strip_sample_count = (tile_sample_count * tiles_across)
+                - (padding_right * tile_length * samples_per_pixel);
+
+            self.tile_attributes = Some(TileAttributes {
+                tile_offsets: self.get_tag_u64_vec(Tag::TileOffsets)?,
+                tile_bytes: self.get_tag_u64_vec(Tag::TileByteCounts)?,
+                tile_width,
+                tile_length,
+                tiles_across,
+                tiles_down,
+                tile_samples: tile_sample_count,
+                padding_right,
+                padding_down: (tiles_down * tile_length) - usize::try_from(self.height)?,
+                padding_buffer: vec![0; padding_right * samples_per_pixel * buffer_byte_len],
```

Comment:
This is allocating a potentially very large buffer where the size is controlled by the input. Please compare the desired length against the `limits`.